src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs

```diff
@@ -22,5 +24,39 @@ public class AzureCosmosDBResource(string name, string? connectionString)
     /// Gets the connection string to use for this database.
     /// </summary>
     /// <returns>The connection string to use for this database.</returns>
-    public string? GetConnectionString() => ConnectionString;
+    public string? GetConnectionString() => IsEmulator
+        ? AzureCosmosDBEmulatorConnectionString.Create(GetEmulatorPort("emulator"))
+        : ConnectionString;
+
+    /// <summary>
+    /// Gets a value indicating whether the Azure Cosmos DB resource is running in the local emulator.
+    /// </summary>
+    public bool IsEmulator => this.IsContainer();
+
+    private int GetEmulatorPort(string endpointName) =>
+        Annotations
+            .OfType<ServiceBindingAnnotation>()
+            .FirstOrDefault(x => x.Name == endpointName)
+            ?.Port
+        ?? throw new DistributedApplicationException($"Azure Cosmos DB resource does not have endpoint annotation with name '{endpointName}'.");
+}
+
+static file class AzureCosmosDBEmulatorConnectionString
```

Comment:
Why isn't this just a normal `private static` method in `AzureCosmosDBResource`? Why make another class just for 1 method?

Reply:
I was following the pattern established in the `AzureStorageResource`:

https://github.com/dotnet/aspire/blob/main/src/Aspire.Hosting.Azure/AzureStorageResource.cs#L55

I'm thinking this might be a bit more involved, when more ports are added. Right now, this is the NoSQL provider, I'm waiting on some of the Azure Cosmos DB folks to get back to me about with regard to which ports should be used for each provider.

Reply:
The connection string format and port numbers change for each API, so eventually a switch may be more appropriate than constructing the connection string using string interpolation.

We don't have a specific table documented, but the emulator command-line syntax page and the emulator how-to guide spell out the ports that are exposed by Docker: https://learn.microsoft.com/en-us/azure/cosmos-db/how-to-develop-emulator?tabs=docker-linux%2Ccsharp&pivots=api-mongodb#start-the-emulator

It's straightforward for these two commonly-used APIs:

For the API for NoSQL: `AccountEndpoint=https://localhost:8081/;AccountKey=C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==`

For the API for MongoDB (RU): mongodb://localhost:C2y6yDjf5%2FR%2Bob0N8A7Cgv30VRDJIWEHLM%2B4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw%2FJw%3D%3D@localhost:10255/admin?ssl=true

Reply:
I opted to support just NoSQL API for now, until I know for sure which APIs are fully supported with the emulator. From what I can tell, it's limited.